lib/storage/src/content_manager/toc.rs

```diff
@@ -541,7 +542,17 @@ impl TableOfContent {
             removed.before_drop().await;
             let path = self.get_collection_path(collection_name);
             drop(removed);
-            remove_dir_all(path).map_err(|err| {
+
+            // Move collection to ".deleted" folder to prevent accidental reuse
+            let uuid = Uuid::new_v4().to_string();
+            let removed_collections_path =
+                Path::new(&self.storage_config.storage_path).join(".deleted");
+            tokio::fs::create_dir_all(&removed_collections_path).await?;
+            let deleted_path = removed_collections_path
+                .join(collection_name)
+                .with_extension(uuid);
+            tokio::fs::rename(path, &deleted_path).await?;
```

Comment:
because the deleted path is not used anywhere, it is possible to remove `deleted_path` at any moment. I propose to delete in background mode and get a response to the user earlier

Reply:
hm, interesting